app/controllers/tasks_controller.rb

```diff
@@ -1,29 +1,31 @@
 class TasksController < ApplicationController
+  before_action :check_for_user, only: [:create]
 
   def new
     @task = Task.new
     @task_categories = TaskCategory.all
   end
 
+
   def create
-    if current_user.nil?
-      session[:task] = task_params
-      redirect_to new_requester_session_path
+    @task = Task.new(task_params)
+    @task.user = current_user
+    @task.task_category_id = params[:task][:task_category_id]
```

Comment:
The variable @task is used in the show view, showing the newly created task (row 15)